deployment.rst

```diff
@@ -1,8 +1,215 @@
+.. index::
+   single: Deployment; Deployment tools
+
+.. _how-to-deploy-a-symfony2-application:
+
 Deployment
 ==========
 
+Deploying a Symfony application can be a complex and varied task depending on
+the setup and the requirements of your application. This article is not a step-
+by-step guide, but is a general list of the most common requirements and ideas
+for deployment.
+
+.. _symfony2-deployment-basics:
+
+Symfony Deployment Basics
+-------------------------
+
+The typical steps taken while deploying a Symfony application include:
+
+#. Upload your code to the production server;
+#. Install your vendor dependencies (typically done via Composer and may be done
+   before uploading);
+#. Running database migrations or similar tasks to update any changed data structures;
+#. Clearing (and optionally, warming up) your cache.
+
+A deployment may also include other tasks, such as:
+
+* Tagging a particular version of your code as a release in your source control
+  repository;
+* Creating a temporary staging area to build your updated setup "offline";
+* Running any tests available to ensure code and/or server stability;
+* Removal of any unnecessary files from the ``web/`` directory to keep your
+  production environment clean;
+* Clearing of external cache systems (like `Memcached`_ or `Redis`_).
+
+How to Deploy a Symfony Application
+-----------------------------------
+
+There are several ways you can deploy a Symfony application. Start with a few
+basic deployment strategies and build up from there.
+
+Basic File Transfer
+~~~~~~~~~~~~~~~~~~~
+
+The most basic way of deploying an application is copying the files manually
+via FTP/SCP (or similar method). This has its disadvantages as you lack control
+over the system as the upgrade progresses. This method also requires you
+to take some manual steps after transferring the files (see `Common Post-Deployment Tasks`_)
+
+Using Source Control
+~~~~~~~~~~~~~~~~~~~~
+
+If you're using source control (e.g. Git or SVN), you can simplify by having
+your live installation also be a copy of your repository. When you're ready
+to upgrade it is as simple as fetching the latest updates from your source
+control system.
+
+This makes updating your files *easier*, but you still need to worry about
+manually taking other steps (see `Common Post-Deployment Tasks`_).
+
+Using Build Scripts and other Tools
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I think we should add a section above this, something like:

> Deploying via a "Platform as a Service"

It would have a sentence or two, and then links to our articles about this, Heroku, platform.sh, etc.